lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -214,6 +214,43 @@ TypeChecker::lookupUnqualifiedType(DeclContext *dc, DeclName name,
                                    NameLookupOptions options) {
   SmallVector<TypeDecl *, 1> decls;
 
+  // CodingKeys is a special type. Although not a protocol requirement of the
+  // Encodable/Decodable protocols, it is synthesized as part of
+  // Encodable/Decodable conformance. Unqualified lookup may occur before the
+  // type is synthesized, though, in which case, we need to make sure it gets
+  // synthesized.
+  //
+  // If this is what we're looking for, then we need to attempt to cause the
+  // type to be synthesized before performing the lookup.
+  if (name.isSimpleName(Context.Id_CodingKeys)) {
```

Comment:
Hm. I don't like that we have to do this before the lookup, but I guess it really might be necessary in case there's a global CodingKeys that would be found otherwise. However… (see below)